data/virt_autotest/fetch_logs_from_guest.sh

```diff
@@ -0,0 +1,260 @@
+      ;;
+    h)
+      help_usage | tee -a ${fetch_logs_from_guest_log}
+      exit 1
+      ;;
+    *)
+      help_usage | tee -a ${fetch_logs_from_guest_log}
+      exit 1
+      ;;
+  esac
+done
+shift "$(($OPTIND -1))"
+if [[ ${virt_logs_folder} == "" ]];then
+   virt_logs_folder="/tmp/virt_logs_residence"
+fi
+if [[ ${virt_guests_wanted} == "" ]];then
+   virt_guests_wanted="all"
+fi
+
+unset guest_hash_ipaddr
+declare -a guest_hash_ipaddr=""
+guest_domain_types="sles"
+guests_inactive_array=`virsh list --inactive | grep -E "${guest_domain_types}" | awk '{print $2}'`
+guest_domains_array=`virsh list  --all | grep -E "${guest_domain_types}" | awk '{print $2}'`
+guest_current=""
+guest_macaddresses_array=""
+guest_ipaddress="";
+guest_hash_index=0
+dhcpd_lease_file="/var/lib/dhcp/db/dhcpd.leases"
+
+#Establish virtual machine domain name and ip address mapping
+for guest_current in ${guest_domains_array[@]};do
+    guest_macaddresses_array[${guest_hash_index}]=`virsh domiflist --domain ${guest_current} | grep -oE "([0-9|a-z]{2}:){5}[0-9|a-z]{2}"`
+    guest_ipaddress=`tac $dhcpd_lease_file | awk '!($0 in S) {print; S[$0]}' | tac | grep -iE "${guest_macaddresses_array[${guest_hash_index}]}" -B8 | grep -oE "([0-9]{1,3}\.){3}[0-9]{1,3}" | tail -1`
+    guest_hash_ipaddr[${guest_hash_index}]=${guest_ipaddress}
+    echo -e ${guest_current}:${guest_hash_ipaddr[${guest_hash_index}]} | tee -a ${virt_logs_collecor_log}
+    guest_hash_index=$(( ${guest_hash_index} + 1 ))
+done
+
+#Start fetching logs from virtual machine
+if [[ ${virt_guests_wanted} == "none" ]];then
+   echo -e "Will not fetch any log from any guest." | tee -a ${fetch_logs_from_guest_log}
+else
+   guest_hash_index=0
+   for guest_current in ${guest_domains_array[@]};do
+       if [[ ${virt_guests_wanted} == "all" ]] || [[ ${virt_guests_wanted} =~ .*${guest_current}.* ]];then
+          if [[ ${guests_inactive_array[@]} == .*${guest_current}.* ]];then
+             echo -e "Virtual machine ${guest_current} in shutdown state. Skip fetching logs from it." | tee -a ${fetch_logs_from_guest_log}
+          else
+             echo -e "fetch_logs_from_guest ${guest_current} ${virt_logs_folder}" | tee -a ${fetch_logs_from_guest_log}
+             fetch_logs_from_guest ${guest_current} ${guest_hash_ipaddr[${guest_hash_index}]} ${virt_logs_folder} ${virt_extra_logs_guest[@]} | tee -a ${fetch_logs_from_guest_log}
+             fetch_logs_from_guest_log_result=$(( ${fetch_logs_from_guest_log_result} | $? ))
+          fi
+       else
+          echo -e "Virtual machine ${guest_current} is not wanted. Skip fetching logs from it." | tee -a ${fetch_logs_from_guest_log}
+       fi
+       guest_hash_index=$(( ${guest_hash_index} + 1 ))
+   done
+fi
+compress_virt_logs_folder ${virt_logs_folder} | tee -a ${fetch_logs_from_guest_log}
+fetch_logs_from_guest_log_result=$(( ${fetch_logs_from_guest_log_result} | $? ))
+exit ${fetch_logs_from_guest_log_result}
```

Comment:
Some lines are not well aligned in this PR. I do not give comment for them one by one, please look through and correct.

Reply:
All lines aligned